### PR TITLE
[MM-41999] Add additional logging for debugging, allow users to change log level

### PR DESCRIPTION
**Pull request by devinbinnie**
### Summary
This PR adds a substantial amount of log statements to various functions in the main process. This should help facilitate debugging some of the stranger issues we've seen in the past from users, especially those that cannot be reproduced by developers.

I've switched to using 6 different log levels instead of the 3 we've been using. This loosely corresponds to this guide:
```
error - Something went wrong, and it's probably a big deal
warn - Something went wrong, but it's likely not a big deal
info - Just want to make sure you know something happened
verbose - Same as info if you care about a bit extra information
debug - Tells you basically everything the app is doing
silly - Same as debug, except it's basically every single function in the app
```
(and yes, the top one is called `silly`. Thanks `electron-log` :P)

The user will be able to switch between those log levels using a dropdown in the Settings window. This way developers can instruct users to turn on more verbose logging to help debug some issues. See screenshots below.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41999

#### Screenshots
image
image

#### Release Note
```release-note
Added a dropdown to select the logging level of the app for developers to help users debug complicated issues.
```

### CODE REVIEW

**Comment by devinbinnie** [comment]
@esethna Added you for wording review.

**Review by Willyfrog — APPROVED** [review]

**Review by M-ZubairAhmed — APPROVED** [review]

File: src/renderer/components/SettingsPage.tsx [hunk 1]
```diff
@@ -801,6 +820,27 @@ export default class SettingsPage extends React.PureComponent<Record<string, nev
...
+                {'Logging level'}
+                <FormControl
+                    style={settingsPage.logLevelInput}
+                    as='select'
+                    id='inputLogLevel'
+                    ref={this.logLevelRef}
+                    value={this.state.logLevel}
+                    onChange={this.handleChangeLogLevel}
+                >
+                    <option value='error'>{'Errors only (error)'}</option>
```
Comment (esethna) [inline src/renderer/components/SettingsPage.tsx:833]: Errors (error)
Errors and Warnings (warn)
Info (info)
Verbose (verbose)
Debug (debug)
Finest (silly)

Logging is helpful for developers and support to isolate issues you may be encountering with the desktop app. Increasing the log level increases disk space usage and can impact performance. We recommend only increasing the log level if you are having issues.

**Review by esethna — CHANGES_REQUESTED** [review]

File: src/renderer/components/SettingsPage.tsx (hunk 1 above)
Comment (esethna) [reply src/renderer/components/SettingsPage.tsx:833]: @devinbinnie thoughts? ^

Comment (devinbinnie) [reply src/renderer/components/SettingsPage.tsx:833]: Works for me, will make the changes.

**Review by esethna — APPROVED** [review]

**Pull request merged**